### PR TITLE
DRIVERS-1293: Remove deprecated terms from transaction specs

### PR DESCRIPTION
**Pull request by JamesKovacs**

### CODE REVIEW

File: source/transactions/tests/legacy/mongos-recovery-token.yml [hunk 1]
```diff
@@ -194,7 +194,7 @@ tests:
         result:
           insertedId: 1
       # Enable the fail point only on the Mongos that session0 is pinned to.
-      # Fail isMaster to prevent the heartbeat requested directly after the
+      # Fail isMaster/hello to prevent the heartbeat requested directly after the
```
Comment (durran) [inline source/transactions/tests/legacy/mongos-recovery-token.yml:197]: Should this be hello/legacy hello?

File: source/transactions/tests/legacy/pin-mongos.yml [hunk 2]
```diff
@@ -461,7 +461,7 @@ tests:
         result:
           insertedId: 3
       # Enable the fail point only on the Mongos that session0 is pinned to.
-      # Fail isMaster to prevent the heartbeat requested directly after the
+      # Fail isMaster/hello to prevent the heartbeat requested directly after the
```
Comment (durran) [inline source/transactions/tests/legacy/pin-mongos.yml:464]: hello/legacy hello

**Review by durran — CHANGES_REQUESTED** [review]

File: source/transactions/tests/legacy/mongos-recovery-token.yml (hunk 1 above)
Comment (JamesKovacs) [reply source/transactions/tests/legacy/mongos-recovery-token.yml]: Good point. Updated.

File: source/transactions/tests/legacy/pin-mongos.yml (hunk 2 above)
Comment (JamesKovacs) [reply source/transactions/tests/legacy/pin-mongos.yml]: Good point. Updated.

**Review by durran — APPROVED** [review]
LGTM

**Review by iwysiu — APPROVED** [review]

**Pull request merged**